controllers/propagator/policyPredicate.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package propagator
+
+import (
+	"k8s.io/apimachinery/pkg/api/equality"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+)
+
+// policyPredicates filters out updates to policies that are pure status updates.
+func policyPredicates() predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// nolint: forcetypeassert
+			oldPolicy := e.ObjectOld.(*policiesv1.Policy)
+			// nolint: forcetypeassert
+			updatedPolicy := e.ObjectNew.(*policiesv1.Policy)
+
+			// Ignore pure status updates since those are handled by a separate controller
+			return !equality.Semantic.DeepEqual(oldPolicy.ObjectMeta.Labels, updatedPolicy.ObjectMeta.Labels) ||
+				!equality.Semantic.DeepEqual(oldPolicy.ObjectMeta.Annotations, updatedPolicy.ObjectMeta.Annotations) ||
+				!equality.Semantic.DeepEqual(oldPolicy.Spec, updatedPolicy.Spec)
```

Comment:
Can we check generation instead? This DeepEqual could be expensive.

Reply:
Great idea! Thanks.